dm/dm/worker/status.go

```diff
@@ -123,3 +124,25 @@ func (w *SourceWorker) GetUnitAndSourceStatusJSON(stName string, sourceStatus *b
 	}
 	return s
 }
+
+func (w *SourceWorker) GetValidateStatus(stName string, filterStatus string) []*pb.ValidationStatus {
+	sts := w.subTaskHolder.getAllSubTasks()
+	res := make([]*pb.ValidationStatus, 0)
+	if len(sts) == 0 {
+		return res
+	}
+	sourceIP := w.cfg.DecryptPassword().From.Host
+	for _, st := range sts {
+		if st.cfg.Name != stName || st.validator == nil || !st.validator.Started() {
+			continue
+		}
+		tblStats := st.validator.GetValidationStatus()
```

Comment:
`st.validator` might be accessed concurrently

prefer to access methods of subtask, not access its field directly.

Reply:
Did you mean I should wrap it with a method and acquire a lock before accessing it?

Reply:
yes, see stopValidator